test/helpers/deploy-helpers.ts

```diff
@@ -0,0 +1,134 @@
+
+// For use in inegration test of deployment campaign
+
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { TZNSContractState } from "../../src/deploy/campaign/types";
+import { BigNumber, ethers } from "ethers";
+import { ICurvePriceConfig, IDistributionConfig } from "./types";
+import { expect } from "chai";
+import { hashDomainLabel } from ".";
+import { getDomainHashFromEvent } from "./events";
+
+export const approveBulk = async (
+  signers : Array<SignerWithAddress>,
+  zns : TZNSContractState,
+) => {
+  for (const signer of signers) {
+    const tx = await zns.meowToken.connect(signer).approve(
+      zns.treasury.address,
+      ethers.constants.MaxUint256,
+    );
+
+    await tx.wait(); // hang on hardhat?
+  }
+};
+
+export const mintBulk = async (
+  signers : Array<SignerWithAddress>,
+  amount : BigNumber,
+  zns : TZNSContractState,
+) => {
+  for (const signer of signers) {
+    await zns.meowToken.connect(signer).mint(
+      signer.address,
+      amount
+    );
+  }
+};
+
+export const getPriceBulk = async (
+  domains : Array<string>,
+  zns : TZNSContractState,
+  parentHashes : Array<string> = [],
+  includeProtocolFee  = false,
+) => {
+  let index = 0;
+  const prices = [];
+
+  for (const domain of domains) {
+    const parent = parentHashes[index] ? parentHashes[index] : ethers.constants.HashZero;
+
+    const { price, stakeFee } = await zns.curvePricer.getPriceAndFee(
+      parent,
+      domain,
+      true,
+    );
+
+    const priceWithFee = price.add(stakeFee);
+
+    if (includeProtocolFee) {
+      const protocolFee = await zns.curvePricer.getFeeForPrice(ethers.constants.HashZero, priceWithFee);
+
+      prices.push(priceWithFee.add(protocolFee));
+    } else {
+      prices.push(priceWithFee);
+    }
+
+
+    index++;
+  }
+
+  return prices;
+};
+
+export const registerRootDomainBulk = async (
+  signers : Array<SignerWithAddress>,
+  domains : Array<string>,
+  domainAddress : string,
+  tokenUri : string,
+  distConfig : IDistributionConfig,
+  priceConfig : ICurvePriceConfig,
+  zns : TZNSContractState,
+) : Promise<void> => {
+  let index = 0;
+
+  // TODO need to return state for outer calling tests, better way to do this?
```

Comment:
what is this for? still needs to be done?

Reply:
No I resolved that it was more meant to be explanatory. I couldn't get the state of the contracts in the scope of the code that calls `registerRootDomainBulk` to reflect the changes made within that function if I used a prototype function like `.map()` or `.reduce()`, I'm sure there is a way but it wouldn't work for me and I went with this instead. Removed the comment